### PR TITLE
feat: child tables, amended boolean, refactor

### PR DESCRIPTION
**Pull request by robproject**
Add boolean parameter to include links for child tables and legend update.
Add boolean parameter to exclude 'amended_from' field.

image

### CODE REVIEW

**Comment by robproject** [comment]
Breaks fetch functionality

*edit* fixed in next commit

**Comment by nikkothari22** [comment]
@robproject We'll review this over the weekend. Thanks! 

Have you checked [commit.frappe.cloud](https://commit.frappe.cloud)?

**Comment by robproject** [comment]
sure thing!

I hadn't seen that before, looks cool

**Comment by sumitjain236** [comment]
Hey @robproject Can You provide any Screenshot or Photo of the erd generated by your code, for better understanding of code

File: frappe_er_generator/frappe_er_generator/er_generator.py [hunk 1]
```diff
@@ -132,11 +136,15 @@ def get_table(data, link_list, doctypes):
             else:
                 table_element_list.append(
                     f'<tr><td port="{field.get("fieldname")}">{field.get("label")}</td></tr>')
-        if field.get("fieldtype") == "Link":
+        if field.get("fieldtype") == "Link" and (amended_from or field.get("fieldname") != "amended_from"):
```
Comment (sumitjain236) [inline frappe_er_generator/frappe_er_generator/er_generator.py]: 1. What if I want to show Linkage of amended_from field by default , amended_from should be True by default.
2. You need to check field.get('options') !== data.doctype , as in some cases fieldname can be different

Comment (robproject) [reply frappe_er_generator/frappe_er_generator/er_generator.py]: 1. Got it, it won't change original functionality now.
2. It was intended to only ignore amended_from. Modified to omit self-links or any link name passed by itself or in an iterable.

File: frappe_er_generator/frappe_er_generator/er_generator.py [hunk 2]
```diff
@@ -136,7 +144,10 @@ def get_table(data, link_list, doctypes, child_tables, amended_from):
             else:
                 table_element_list.append(
                     f'<tr><td port="{field.get("fieldname")}">{field.get("label")}</td></tr>')
-        if field.get("fieldtype") == "Link" and (amended_from or field.get("fieldname") != "amended_from"):
+        if field.get("fieldtype") == "Link" and (
+        omit_links is None or 
+        (omit_links == 'all' and field.get("options") != data.get("name")) or
```
Comment (sumitjain236) [inline frappe_er_generator/frappe_er_generator/er_generator.py:160]: Again this code have problem.
for eg: I want to omit fieldname `email` from doctype `User` but that field `email` exist in multiple doctype , so according to this code `email` field will be omited in each table where it exist.

Comment (robproject) [reply frappe_er_generator/frappe_er_generator/er_generator.py:160]: Again, that is the intended functionality. I just want to be able to remove amended_from fields while keeping everything else. So then it should be able to specify, per doctype, what fields to omit? Including an 'all' option for both doctype and docfield?

**Pull request closed without merging**